### ISSUE TITLE
fiona currently broken on windows

### ISSUE THREAD

**Issue by gboeing** [conda-forge/fiona-feedstock#153]
Cannot import fiona after installing it in its own conda env with conda-forge:

```
>>> import fiona
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Anaconda\envs\fiona\lib\site-packages\fiona\__init__.py", line 84, in <module>
    from fiona.collection import BytesCollection, Collection
  File "C:\Anaconda\envs\fiona\lib\site-packages\fiona\collection.py", line 9, in <module>
    from fiona.ogrext import Iterator, ItemsIterator, KeysIterator
ImportError: DLL load failed while importing ogrext: The specified module could not be found.
```

Environment (<code>conda list</code>):
<details>

```
$ conda list
(fiona) C:\>conda list
# packages in environment at C:\Anaconda\envs\fiona:
#
# Name                    Version                   Build  Channel
attrs                     19.3.0                     py_0    conda-forge
boost-cpp                 1.70.0               h6a4c333_2    conda-forge
bzip2                     1.0.8                hfa6e2cd_1    conda-forge
ca-certificates           2019.9.11            hecc5488_0    conda-forge
certifi                   2019.9.11                py38_0    conda-forge
cfitsio                   3.470                hfa6e2cd_2    conda-forge
click                     7.0                        py_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.5.0                      py_0    conda-forge
curl                      7.65.3               h4496350_0    conda-forge
expat                     2.2.5             he025d50_1004    conda-forge
fiona                     1.8.11           py38hb7fdc2d_0    conda-forge
freetype                  2.10.0               h563cfd7_1    conda-forge
freexl                    1.0.5             hd288d7e_1002    conda-forge
gdal                      3.0.2            py38h2fee047_4    conda-forge
geos                      3.7.2                he025d50_2    conda-forge
geotiff                   1.5.1                h765a46f_7    conda-forge
gettext                   0.19.8.1          hb01d8f6_1002    conda-forge
glib                      2.58.3          py38hc0c2ac7_1002    conda-forge
hdf4                      4.2.13            hf8e6fe8_1003    conda-forge
hdf5                      1.10.5          nompi_ha405e13_1104    conda-forge
icu                       64.2                 he025d50_1    conda-forge
intel-openmp              2019.4                      245    defaults
jpeg                      9c                hfa6e2cd_1001    conda-forge
kealib                    1.4.10            hf7dc31f_1005    conda-forge
krb5                      1.16.3            hdd46e55_1001    conda-forge
libblas                   3.8.0                    14_mkl    conda-forge
libcblas                  3.8.0                    14_mkl    conda-forge
libcurl                   7.65.3               h4496350_0    conda-forge
libffi                    3.2.1             h6538335_1006    conda-forge
libgdal                   3.0.2                h4a92af1_4    conda-forge
libiconv                  1.15              hfa6e2cd_1005    conda-forge
libkml                    1.3.0             h4ece8bf_1010    conda-forge
liblapack                 3.8.0                    14_mkl    conda-forge
libnetcdf                 4.7.1           nompi_he2c6080_102    conda-forge
libpng                    1.6.37               h7602738_0    conda-forge
libpq                     11.5                 hb0bdaea_2    conda-forge
libspatialite             4.3.0a            h2b2ca8d_1032    conda-forge
libssh2                   1.8.2                h642c060_2    conda-forge
libtiff                   4.1.0                h21b02b4_1    conda-forge
libwebp                   1.0.2                hfa6e2cd_4    conda-forge
libxml2                   2.9.10               h9ce36c8_0    conda-forge
lz4-c                     1.8.3             he025d50_1001    conda-forge
m2w64-expat               2.1.1                         2    defaults
m2w64-gcc-libgfortran     5.3.0                         6    defaults
m2w64-gcc-libs            5.3.0                         7    defaults
m2w64-gcc-libs-core       5.3.0                         7    defaults
m2w64-gettext             0.19.7                        2    defaults
m2w64-gmp                 6.1.0                         2    defaults
m2w64-libiconv            1.14                          6    defaults
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    defaults
m2w64-xz                  5.2.2                         2    defaults
mkl                       2019.4                      245    defaults
msys2-conda-epoch         20160418                      1    defaults
munch                     2.5.0                      py_0    conda-forge
numpy                     1.17.3           py38hc71023c_0    conda-forge
openjpeg                  2.3.1                h57dd2e7_2    conda-forge
openssl                   1.1.1d               hfa6e2cd_0    conda-forge
pcre                      8.43                 h6538335_0    conda-forge
pip                       19.3.1                   py38_0    conda-forge
poppler                   0.67.0               h1707e21_8    conda-forge
poppler-data              0.4.9                         1    conda-forge
postgresql                11.5                 h06f7779_2    conda-forge
proj                      6.2.1                ha7a8c7b_0    conda-forge
python                    3.8.0                hc9e8b01_5    conda-forge
setuptools                42.0.1                   py38_0    conda-forge
shapely                   1.6.4           py38ha35856d_1006    conda-forge
six                       1.13.0                   py38_0    conda-forge
sqlite                    3.30.1               hfa6e2cd_0    conda-forge
tbb                       2018.0.5             he980bc4_0    conda-forge
tiledb                    1.7.0                hffbbd95_2    conda-forge
tk                        8.6.10               hfa6e2cd_0    conda-forge
vc                        14.1                 h0510ff6_4    defaults
vs2015_runtime            14.16.27012          hf0eaf9b_0    defaults
wheel                     0.33.6                   py38_0    conda-forge
wincertstore              0.2                   py38_1003    conda-forge
xerces-c                  3.2.2             h6538335_1004    conda-forge
xz                        5.2.4             h2fa13f4_1001    conda-forge
zlib                      1.2.11            h2fa13f4_1006    conda-forge
zstd                      1.4.4                hd8a0e53_1    conda-forge
```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info
C:\>conda info

     active environment : None
            shell level : 0
       user config file : C:\Users\Geoff\.condarc
 populated config files : C:\Users\Geoff\.condarc
          conda version : 4.7.12
    conda-build version : 3.18.11
         python version : 3.7.3.final.0
       virtual packages : __cuda=10.1
       base environment : C:\Anaconda  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Anaconda\pkgs
                          C:\Users\Geoff\.conda\pkgs
                          C:\Users\Geoff\AppData\Local\conda\conda\pkgs
       envs directories : C:\Anaconda\envs
                          C:\Users\Geoff\.conda\envs
                          C:\Users\Geoff\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.7.12 requests/2.22.0 CPython/3.7.3 Windows/10 Windows/10.0.18362
          administrator : False
             netrc file : None
           offline mode : False
```
</details>

**Comment by ocefpaf** [comment]
We had a couple of migrations happening in the past few days but all that have to do with `gdal` ended. I just tried this and I cannot produce it. Here is the list of packages I got:

```
attrs              conda-forge/noarch::attrs-19.3.0-py_0
  boost-cpp          conda-forge/win-64::boost-cpp-1.70.0-h6a4c333_2
  bzip2              conda-forge/win-64::bzip2-1.0.8-hfa6e2cd_1
  ca-certificates    conda-forge/win-64::ca-certificates-2019.11.28-hecc5488_0
  certifi            conda-forge/win-64::certifi-2019.11.28-py38_0
  cfitsio            conda-forge/win-64::cfitsio-3.470-hfa6e2cd_2
  click              conda-forge/noarch::click-7.0-py_0
  click-plugins      conda-forge/noarch::click-plugins-1.1.1-py_0
  cligj              conda-forge/noarch::cligj-0.5.0-py_0
  curl               conda-forge/win-64::curl-7.65.3-h4496350_0
  expat              conda-forge/win-64::expat-2.2.5-he025d50_1004
  fiona              conda-forge/win-64::fiona-1.8.11-py38hb7fdc2d_0
  freetype           conda-forge/win-64::freetype-2.10.0-h563cfd7_1
  freexl             conda-forge/win-64::freexl-1.0.5-hd288d7e_1002
  gdal               conda-forge/win-64::gdal-3.0.2-py38h2fee047_5
  geos               conda-forge/win-64::geos-3.8.0-he025d50_0
  geotiff            conda-forge/win-64::geotiff-1.5.1-h765a46f_7
  gettext            conda-forge/win-64::gettext-0.19.8.1-hb01d8f6_1002
  glib               conda-forge/win-64::glib-2.58.3-py38hc0c2ac7_1002
  hdf4               conda-forge/win-64::hdf4-4.2.13-hf8e6fe8_1003
  hdf5               conda-forge/win-64::hdf5-1.10.5-nompi_ha405e13_1104
  icu                conda-forge/win-64::icu-64.2-he025d50_1
  intel-openmp       pkgs/main/win-64::intel-openmp-2019.4-245
  jpeg               conda-forge/win-64::jpeg-9c-hfa6e2cd_1001
  kealib             conda-forge/win-64::kealib-1.4.10-hf7dc31f_1005
  krb5               conda-forge/win-64::krb5-1.16.3-hdd46e55_1001
  libblas            conda-forge/win-64::libblas-3.8.0-14_mkl
  libcblas           conda-forge/win-64::libcblas-3.8.0-14_mkl
  libcurl            conda-forge/win-64::libcurl-7.65.3-h4496350_0
  libffi             conda-forge/win-64::libffi-3.2.1-h6538335_1006
  libgdal            conda-forge/win-64::libgdal-3.0.2-hafedd80_5
  libiconv           conda-forge/win-64::libiconv-1.15-hfa6e2cd_1005
  libkml             conda-forge/win-64::libkml-1.3.0-h4ece8bf_1010
  liblapack          conda-forge/win-64::liblapack-3.8.0-14_mkl
  libnetcdf          conda-forge/win-64::libnetcdf-4.7.1-nompi_he2c6080_102
  libpng             conda-forge/win-64::libpng-1.6.37-h7602738_0
  libpq              conda-forge/win-64::libpq-11.5-hb0bdaea_2
  libspatialite      conda-forge/win-64::libspatialite-4.3.0a-he9c6e69_1033
  libssh2            conda-forge/win-64::libssh2-1.8.2-h642c060_2
  libtiff            conda-forge/win-64::libtiff-4.1.0-h21b02b4_1
  libwebp            conda-forge/win-64::libwebp-1.0.2-hfa6e2cd_4
  libxml2            conda-forge/win-64::libxml2-2.9.10-h9ce36c8_0
  lz4-c              conda-forge/win-64::lz4-c-1.8.3-he025d50_1001
  m2w64-expat        pkgs/msys2/win-64::m2w64-expat-2.1.1-2
  m2w64-gcc-libgfor~ pkgs/msys2/win-64::m2w64-gcc-libgfortran-5.3.0-6
  m2w64-gcc-libs     pkgs/msys2/win-64::m2w64-gcc-libs-5.3.0-7
  m2w64-gcc-libs-co~ pkgs/msys2/win-64::m2w64-gcc-libs-core-5.3.0-7
  m2w64-gettext      pkgs/msys2/win-64::m2w64-gettext-0.19.7-2
  m2w64-gmp          pkgs/msys2/win-64::m2w64-gmp-6.1.0-2
  m2w64-libiconv     pkgs/msys2/win-64::m2w64-libiconv-1.14-6
  m2w64-libwinpthre~ pkgs/msys2/win-64::m2w64-libwinpthread-git-5.0.0.4634.697f757-2
  m2w64-xz           pkgs/msys2/win-64::m2w64-xz-5.2.2-2
  mkl                pkgs/main/win-64::mkl-2019.4-245
  msys2-conda-epoch  pkgs/msys2/win-64::msys2-conda-epoch-20160418-1
  munch              conda-forge/noarch::munch-2.5.0-py_0
  numpy              conda-forge/win-64::numpy-1.17.3-py38hc71023c_0
  openjpeg           conda-forge/win-64::openjpeg-2.3.1-h57dd2e7_2
  openssl            conda-forge/win-64::openssl-1.1.1d-hfa6e2cd_0
  pcre               conda-forge/win-64::pcre-8.43-h6538335_0
  poppler            conda-forge/win-64::poppler-0.67.0-h1707e21_8
  poppler-data       conda-forge/noarch::poppler-data-0.4.9-1
  postgresql         conda-forge/win-64::postgresql-11.5-h06f7779_2
  proj               conda-forge/win-64::proj-6.2.1-ha7a8c7b_0
  python             conda-forge/win-64::python-3.8.0-hc9e8b01_5
  setuptools         conda-forge/win-64::setuptools-42.0.1-py38_0
  shapely            conda-forge/win-64::shapely-1.6.4-py38h2130f3d_1007
  six                conda-forge/win-64::six-1.13.0-py38_0
  sqlite             conda-forge/win-64::sqlite-3.30.1-hfa6e2cd_0
  tbb                conda-forge/win-64::tbb-2018.0.5-he980bc4_0
  tiledb             conda-forge/win-64::tiledb-1.7.0-hffbbd95_2
  tk                 conda-forge/win-64::tk-8.6.10-hfa6e2cd_0
  vc                 pkgs/main/win-64::vc-14.1-h0510ff6_4
  vs2015_runtime     pkgs/main/win-64::vs2015_runtime-14.16.27012-hf0eaf9b_0
  wincertstore       conda-forge/win-64::wincertstore-0.2-py38_1003
  xerces-c           conda-forge/win-64::xerces-c-3.2.2-h6538335_1004
  xz                 conda-forge/win-64::xz-5.2.4-h2fa13f4_1001
  zlib               conda-forge/win-64::zlib-1.2.11-h2fa13f4_1006
  zstd               conda-forge/win-64::zstd-1.4.4-hd8a0e53_1
```

Note I got the `gdal `build number 5.

**Comment by gboeing** [comment]
@ocefpaf I'm out of town, but I have just re-built the conda env on a different windows computer to test again. It is still producing a broken environment (even with gdal build number 5 now).

```
C:\Users\Geoff>conda create -n fiona fiona

C:\Users\Geoff>conda activate fiona

(fiona) C:\Users\Geoff>conda list
# packages in environment at c:\Anaconda\envs\fiona:
#
# Name                    Version                   Build  Channel
attrs                     19.3.0                     py_0    conda-forge
boost-cpp                 1.70.0               h6a4c333_2    conda-forge
bzip2                     1.0.8                hfa6e2cd_1    conda-forge
ca-certificates           2019.11.28           hecc5488_0    conda-forge
certifi                   2019.11.28               py38_0    conda-forge
cfitsio                   3.470                hfa6e2cd_2    conda-forge
click                     7.0                        py_0    conda-forge
click-plugins             1.1.1                      py_0    conda-forge
cligj                     0.5.0                      py_0    conda-forge
curl                      7.65.3               h4496350_0    conda-forge
expat                     2.2.5             he025d50_1004    conda-forge
fiona                     1.8.11           py38hb7fdc2d_0    conda-forge
freetype                  2.10.0               h563cfd7_1    conda-forge
freexl                    1.0.5             hd288d7e_1002    conda-forge
gdal                      3.0.2            py38h2fee047_5    conda-forge
geos                      3.8.0                he025d50_0    conda-forge
geotiff                   1.5.1                h765a46f_7    conda-forge
gettext                   0.19.8.1          hb01d8f6_1002    conda-forge
glib                      2.58.3          py38hc0c2ac7_1002    conda-forge
hdf4                      4.2.13            hf8e6fe8_1003    conda-forge
hdf5                      1.10.5          nompi_ha405e13_1104    conda-forge
icu                       64.2                 he025d50_1    conda-forge
intel-openmp              2019.4                      245    defaults
jpeg                      9c                hfa6e2cd_1001    conda-forge
kealib                    1.4.10            hf7dc31f_1005    conda-forge
krb5                      1.16.3            hdd46e55_1001    conda-forge
libblas                   3.8.0                    14_mkl    conda-forge
libcblas                  3.8.0                    14_mkl    conda-forge
libcurl                   7.65.3               h4496350_0    conda-forge
libffi                    3.2.1             h6538335_1006    conda-forge
libgdal                   3.0.2                hafedd80_5    conda-forge
libiconv                  1.15              hfa6e2cd_1005    conda-forge
libkml                    1.3.0             h4ece8bf_1010    conda-forge
liblapack                 3.8.0                    14_mkl    conda-forge
libnetcdf                 4.7.1           nompi_he2c6080_102    conda-forge
libpng                    1.6.37               h7602738_0    conda-forge
libpq                     11.5                 hb0bdaea_2    conda-forge
libspatialite             4.3.0a            he9c6e69_1033    conda-forge
libssh2                   1.8.2                h642c060_2    conda-forge
libtiff                   4.1.0                h21b02b4_1    conda-forge
libwebp                   1.0.2                hfa6e2cd_4    conda-forge
libxml2                   2.9.10               h9ce36c8_0    conda-forge
lz4-c                     1.8.3             he025d50_1001    conda-forge
m2w64-expat               2.1.1                         2    defaults
m2w64-gcc-libgfortran     5.3.0                         6    defaults
m2w64-gcc-libs            5.3.0                         7    defaults
m2w64-gcc-libs-core       5.3.0                         7    defaults
m2w64-gettext             0.19.7                        2    defaults
m2w64-gmp                 6.1.0                         2    defaults
m2w64-libiconv            1.14                          6    defaults
m2w64-libwinpthread-git   5.0.0.4634.697f757               2    defaults
m2w64-xz                  5.2.2                         2    defaults
mkl                       2019.4                      245    defaults
msys2-conda-epoch         20160418                      1    defaults
munch                     2.5.0                      py_0    conda-forge
numpy                     1.17.3           py38hc71023c_0    conda-forge
openjpeg                  2.3.1                h57dd2e7_2    conda-forge
openssl                   1.1.1d               hfa6e2cd_0    conda-forge
pcre                      8.43                 h6538335_0    conda-forge
pip                       19.3.1                   py38_0    conda-forge
poppler                   0.67.0               h1707e21_8    conda-forge
poppler-data              0.4.9                         1    conda-forge
postgresql                11.5                 h06f7779_2    conda-forge
proj                      6.2.1                ha7a8c7b_0    conda-forge
python                    3.8.0                hc9e8b01_5    conda-forge
setuptools                42.0.1                   py38_0    conda-forge
shapely                   1.6.4           py38h2130f3d_1007    conda-forge
six                       1.13.0                   py38_0    conda-forge
sqlite                    3.30.1               hfa6e2cd_0    conda-forge
tbb                       2018.0.5             he980bc4_0    conda-forge
tiledb                    1.7.0                hffbbd95_2    conda-forge
tk                        8.6.10               hfa6e2cd_0    conda-forge
vc                        14.1                 h0510ff6_4    defaults
vs2015_runtime            14.16.27012          hf0eaf9b_0    defaults
wheel                     0.33.6                   py38_0    conda-forge
wincertstore              0.2                   py38_1003    conda-forge
xerces-c                  3.2.2             h6538335_1004    conda-forge
xz                        5.2.4             h2fa13f4_1001    conda-forge
zlib                      1.2.11            h2fa13f4_1006    conda-forge
zstd                      1.4.4                hd8a0e53_1    conda-forge

(fiona) C:\Users\Geoff>conda info

     active environment : fiona
    active env location : c:\Anaconda\envs\fiona
            shell level : 1
       user config file : C:\Users\Geoff\.condarc
 populated config files : C:\Users\Geoff\.condarc
          conda version : 4.7.11
    conda-build version : 3.18.9
         python version : 3.7.3.final.0
       virtual packages :
       base environment : c:\Anaconda  (writable)
           channel URLs : https://conda.anaconda.org/conda-forge/win-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : c:\Anaconda\pkgs
                          C:\Users\Geoff\.conda\pkgs
                          C:\Users\Geoff\AppData\Local\conda\conda\pkgs
       envs directories : c:\Anaconda\envs
                          C:\Users\Geoff\.conda\envs
                          C:\Users\Geoff\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.7.11 requests/2.22.0 CPython/3.7.3 Windows/10 Windows/10.0.18362
          administrator : False
             netrc file : None
           offline mode : False
```

FWIW here's my .condarc:
```
report_errors: false
show_channel_urls: true
channel_priority: strict
channels:
  - conda-forge
  - defaults
```

**Comment by gboeing** [comment]
I should have clarified, the fiona import error is different now (perhaps due to different gdal build?):
```
>>> import fiona
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "c:\Anaconda\envs\fiona\lib\site-packages\fiona\__init__.py", line 84, in <module>
    from fiona.collection import BytesCollection, Collection
  File "c:\Anaconda\envs\fiona\lib\site-packages\fiona\collection.py", line 9, in <module>
    from fiona.ogrext import Iterator, ItemsIterator, KeysIterator
ImportError: DLL load failed while importing ogrext: The specified module could not be found.
```

**Comment by ocefpaf** [comment]
Still working for me. (I updated the list above with a fresh env today.)
Maybe you have something cached? Can you try `conda clean --all --yes` and then re-create the env?

**Comment by gboeing** [comment]
Thanks! Got it solved, but with one caveat that I'll document here for future reference.

I had previously tried `conda clean --all --yes` before creating the conda env. I tried again today, and the env was still broken upon creation. So I deleted the env and then ran `conda clean --all --yes` *twice*. The second time it ran, it still found a few packages to clean, suggesting that a single run did not clean up everything completely. After the second run, I ran it a third time and confirmed there were no more packages to clean. At this point I created the env, and all worked perfectly.
